tests/test_runner.py

```diff
@@ -65,6 +65,24 @@
             {},
             id="no-type-dict",
         ),
+        *(
+            [
+                pytest.param(
+                    "tests.script_runner.parameter_inputs:str_or_int_parameter",
+                    [{"name": "my_str_or_int", "value": "hi there"}],
+                    "type given: str",
+                    id="str-or-int-given-str",
+                ),
+                pytest.param(
+                    "tests.script_runner.parameter_inputs:str_or_int_parameter",
+                    [{"name": "my_str_or_int", "value": "3"}],
+                    "type given: int",
+                    id="str-or-int-given-int",
+                ),
+            ]
+            if _PYDANTIC_VERSION > 1
```

Comment:
The `parse_obj_as` function we use for Pydantic v1 does not appear to correctly handle `Union[int, str]`; it is serializing ints back to strs: https://github.com/argoproj-labs/hera/blob/e7ed95b692fad12ae49bc2598116d89dc5ee8abd/src/hera/workflows/_runner/util.py#L103

Rather than dig into this, I've disabled these two tests on the Pydantic v1 suite for now.

Reply:
Probably related to https://docs.pydantic.dev/1.10/usage/model_config/#smart-union

Reply:
Amazing! Fixed, thanks.